autogpt_platform/backend/backend/data/credit.py

```diff
@@ -107,8 +107,24 @@ async def get_or_refill_credit(self, user_id: str) -> int:
     def time_now():
         return datetime.now(timezone.utc)
 
-    @staticmethod
+    def _is_filter_match(self, cost_filter: BlockInput, input_data: BlockInput) -> bool:
+        """
+        Filter rules:
+          - If costFilter is an object, then check if costFilter is the subset of inputValues
+          - Otherwise, check if costFilter is equal to inputValues.
+          - Undefined, null, and empty string are considered as equal.
+        """
+        if not isinstance(cost_filter, dict) or not isinstance(input_data, dict):
+            return cost_filter == input_data
+
+        return all(
+            (not input_data.get(k) and not v)
+            or (input_data.get(k) and self._is_filter_match(v, input_data[k]))
+            for k, v in cost_filter.items()
+        )
```

Comment:
nit: put this below `_block_usage_cost` since that is the only method using this one

Reply:
Updated